box-browse-sdk/src/main/java/com/box/androidsdk/browse/uidata/ThumbnailManager.java

```diff
@@ -530,18 +535,66 @@ public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
                 public void run() {
                     //TODO decide whether to use two views for crossfading animation.
                     imageView.setImageBitmap(bitmap);
+                    ViewData.getImageLoadListener(imageView).onSuccess();
                 }
             });
         }
 
     }
 
-
-
     private boolean isRequestStillApplicable(final BoxRequest request, final ImageView view) {
         return view.getDrawable() instanceof LoaderDrawable &&
                ((LoaderDrawable) view.getDrawable()).matchesRequest(request);
 
     }
 
+    /**
+     * Interface to listen for Image load, so that the calling component can
+     *  act upon error or success
+     */
+    public interface ImageLoadListener {
+        void onSuccess();
+        void onError();
+    }
+
+    /**
+     * Internal class to keep ImageView data information, like type of thumbnail to show, and image load listener
+     */
+    private static class ViewData {
+        private ImageLoadListener mListener;
+        private String mImageType;
+
+        ViewData(String type, ImageLoadListener listener) {
+            mImageType = type;
+            mListener = listener;
+        }
+
+        static ImageLoadListener getImageLoadListener(ImageView view) {
+            if(view != null) {
+                ViewData data = (ViewData) view.getTag();
+                if(data != null) {
+                    return data.mListener;
+                }
+            }
+            // Dummy listener, so the caller does not need to worry about null checks (Null object pattern)
+            return new ImageLoadListener() {
```

Comment:
Not a big difference but usually they do:

private static ViewData NULL_OBJECT = new ImageLoadListener() {
  
            
  Write  Preview

Not a big difference but usually they do:
```java
private static final ViewData NULL_OBJECT = new ImageLoadListener() {
  @Override
 public void onSuccess() {}
 
  @Override
   public void onError() {}
};
```
and then 
```java
return NULL_OBJECT;

Reply:
done